### PR TITLE
Removing the warning supressions for CS7035 and BC42366

### PR DESCRIPTION
**Pull request by tannergooding**
This updates the Build and File version to be compliant (no part of the version is greater than 65535).

This updates the Informational version (which is just a string) to give more detailed information.

This updates the GetBuildVersion task to return a more detailed build version for the VSIXs, so that two different builds will have two different manifest versions (the VSIX version is required to conform to System.Version, which allows each part to be in the range of 0 through `int.Max`).

### CODE REVIEW

**Comment by tannergooding** [comment]
FYI. @jaredpar, @jasonmalinowski, @Pilchie

**Comment by tannergooding** [comment]
Also, @jmarolf

**Comment by tannergooding** [comment]
Also, FYI. @dotnet/roslyn-infrastructure

File: build/Targets/VSL.Versions.targets [hunk 1]
```diff
@@ -73,7 +73,7 @@
   </Choose>
 
   <!-- Returns the current build version. Used in .vsixmanifests to substitute our build version into them -->
-  <Target Name="GetBuildVersion" Outputs="$(BuildVersion)" />
+  <Target Name="GetBuildVersion" Outputs="$(RoslynSemanticVersion).$(BuildNumberPart1.Trim())$(BuildNumberPart2.Trim())" />
```
Comment (jasonmalinowski) [inline build/Targets/VSL.Versions.targets]: Why the Trim()s? Shouldn't `BuildNumberPart1` and `BuildNumberPart2` come pre-trimmed?

File: build/Targets/Settings.targets [hunk 2]
```diff
@@ -243,15 +243,6 @@
...
         <VBRuntime>Embed</VBRuntime>
-
-        <!-- Suppress the following warnings by default for C# projects
-                42366: GeneratedAssemblyInfo_<version>.vb will report this warning if any of the four parts
-                       of the version string (major.minor.build.revision) are above 65535. This breaks for
-                       various build formats involving the current year/month/day. For example it works for
-                       60614 (June 14th, 2016), but breaks for 70614 (June 14th, 2017) and it also breaks
-                       for something like 20160614 (also June 14th, 2016).
-        -->
-        <NoWarn>$(NoWarn);42366</NoWarn>
```
Comment (jasonmalinowski) [inline build/Targets/Settings.targets]: Can we now do the reverse of switching this to warn-as-error?

File: build/Targets/VSL.Versions.targets (hunk 1 above)
Comment (jasonmalinowski) [inline build/Targets/VSL.Versions.targets]: Also, what _is_ `BuildNumberPart1` and `BuildNumberPart2`? Why were they separated? Are they lonely, which is why we're putting them back together?

**Comment by jasonmalinowski** [comment]
Conceptually seems fine, but I'd appreciate some renaming of some of the variables to something that describes their intent. This is very hard to follow what's going on.

File: build/Targets/VSL.Versions.targets (hunk 1 above)
Comment (tannergooding) [inline build/Targets/VSL.Versions.targets]: They were here long before my change, the parts are just x.y and they are put together differently in some scenarios (x-y for nuget per build pre-release, xy for vsix version, etc)

File: build/Targets/Settings.targets (hunk 2 above)
Comment (tannergooding) [inline build/Targets/Settings.targets]: We currently have everything as 'WarnAsError' during our builds

File: build/Targets/VSL.Versions.targets (hunk 1 above)
Comment (tannergooding) [inline build/Targets/VSL.Versions.targets]: As for the Trim(), it's probably some legacy support for the old signed-build system/TFS and it may or may not be needed.

**Comment by tannergooding** [comment]
@jasonmalinowski, which variables are confusing (everything except for `BuildNumberPart1/2` seems pretty straightforward, and I'm not sure of better names for those variables)?

**Comment by jasonmalinowski** [comment]
`BuildNumberPart1/2` are confusing to me, since Part2 is doing some various trimming and padding and such. Should we just call that `BuildNumberymmdd` or something?

**Comment by tannergooding** [comment]
So our signed build system gives out numbers in the format of yyyymmdd.nn, where the first part (BuildNumberPart1) is something like 20160615 and the second part (BuildNumberPart2) is the nth build that was kicked off that day. So for the first build of the day we get 20160615.1 and for the 52nd build we get 20160616.52

**Comment by tannergooding** [comment]
I'm just going to add comments explaining what each part represents for the time being, this is technically blocking the signed builds and probably shouldn't be held up due to the names of variables that have been around for over a year.

**Comment by tannergooding** [comment]
Comment has been added.

**Comment by tannergooding** [comment]
@jasonmalinowski, am I good to merge once the tests are passing again?

**Comment by tannergooding** [comment]
test windows_debug_unit32_prtest please

File: build/Targets/VSL.Versions.targets (hunk 1 above)
Comment (jasonmalinowski) [inline build/Targets/VSL.Versions.targets]: Let's drop the Trim(). I can't imagine any legacy system that needed that either.

File: build/Targets/VSL.Versions.targets [hunk 3]
```diff
@@ -56,7 +59,7 @@
     <When Condition="'$(OfficialBuild)' == 'true' OR '$(SignedBuild)' == 'true'">
       <PropertyGroup>
         <AssemblyVersion>$(RoslynSemanticVersion).0</AssemblyVersion>
-        <BuildVersion>$(RoslynSemanticVersion).$(BuildNumberPart1)$(BuildNumberPart2)</BuildVersion>
+        <BuildVersion>$(RoslynSemanticVersion).0</BuildVersion>
```
Comment (jasonmalinowski) [inline build/Targets/VSL.Versions.targets]: Doesn't this mean the file version no longer depends on the date? That's not good if it is.

**Comment by jasonmalinowski** [comment]
Comment is awesome, but still more questions, sorry.

File: build/Targets/VSL.Versions.targets (hunk 3 above)
Comment (tannergooding) [inline build/Targets/VSL.Versions.targets]: That was the case (on MBv2) before my original change (unless a build number was manually specified). On MBv1, the existing setup we had would also have broken next year (2017), as they were in the format of ymmdd (so 60615).

**Comment by tannergooding** [comment]
Updated to match the file version produced by MicroBuild v1 and additionally logged #12038 to track getting a proper fix in the next 6 months.

**Comment by tannergooding** [comment]
test windows_eta_open_prtest please

**Comment by jasonmalinowski** [comment]
:+1:
